internal/jimm/applicationoffer.go

```diff
@@ -599,223 +586,115 @@ func (j *JIMM) FindApplicationOffers(ctx context.Context, user *openfga.User, fi
 		return nil, errors.E(op, errors.CodeBadRequest, "at least one filter must be specified")
 	}
 
-	offerFilters, err := j.applicationOfferFilters(ctx, filters...)
-	if err != nil {
-		return nil, errors.E(op, err)
-	}
-	userOfferUUIDs, err := user.ListApplicationOffers(ctx, ofganames.ReaderRelation)
-	if err != nil {
-		return nil, errors.E(op, err)
-	}
-	offerFilters = append(offerFilters, db.ApplicationOfferFilterByUUID(userOfferUUIDs))
-	offers, err := j.Database.FindApplicationOffers(ctx, offerFilters...)
+	// first we find all application offers user has access to
+	offerUUIDs, err := user.ListApplicationOffers(ctx, ofganames.ReaderRelation)
```

Comment:
I wonder if we shouldn't just query all controllers after reading this.
It's either:
1. Query OpenFGA for all offers a user can access.
2. Query DB for each offer to get the controller of the offer.
3. Query the controllers with the user provided filters.
4. Check the user has access to the returned offers.

Versus

1. Query all controllers with the user provided filters.
2. Check the user has access to the returned offers.

Reply:
It's worth it to keep `ListApplicationOffers` as is since that is just querying a single model. But I would simplify `FindApplicationOffers` to just query across all the controllers with the provided filters.

Reply:
done.